modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -5761,6 +5761,22 @@ GDScriptParser::DataType GDScriptAnalyzer::type_from_property(const PropertyInfo
 	return result;
 }
 
+void GDScriptAnalyzer::update_dict_array_return_type(GDScriptParser::DataType p_base_type, const StringName &p_function_name, GDScriptParser::DataType &r_return_type) {
+	const HashSet<String> dict_array_returning_methods = {
+		"keys",
+		"values"
+	};
+
+	if (dict_array_returning_methods.has(p_function_name)) {
+		r_return_type.builtin_type = Variant::ARRAY;
+		if (p_function_name == "keys") {
+			r_return_type.container_element_types = { p_base_type.get_container_element_type_or_variant(0) };
+		} else if (p_function_name == "values") {
+			r_return_type.container_element_types = { p_base_type.get_container_element_type_or_variant(1) };
+		}
+	}
```

Comment:
```suggestion
	if (p_function_name == "keys") {
		r_return_type.builtin_type = Variant::ARRAY;
		r_return_type.container_element_types = { p_base_type.get_container_element_type_or_variant(0) };
	} else if (p_function_name == "values") {
		r_return_type.builtin_type = Variant::ARRAY;
		r_return_type.container_element_types = { p_base_type.get_container_element_type_or_variant(1) };
	}
```
Unnecessary to use a map and then just comparing with the value directly after